src/mlmc/tool/hdf5.py

```diff
@@ -368,16 +369,35 @@ def collected(self, i_chunk=0, chunk_size=512000000, n_samples=None):
                 return dataset[:n_samples]
 
             if chunk_size is not None:
-                if self._items_in_chunk is None:
+                if self.n_items_in_chunk is None:
                     first_item = dataset[0]
                     item_byte_size = first_item.size * first_item.itemsize
-                    self._items_in_chunk = int(np.ceil(chunk_size / item_byte_size))
-                return dataset[i_chunk * self._items_in_chunk: (i_chunk + 1) * self._items_in_chunk]
+                    self.n_items_in_chunk = int(np.ceil(chunk_size / item_byte_size)) \
+                        if int(np.ceil(chunk_size / item_byte_size)) < len(dataset[()]) else len(dataset[()])
+                self._chunks_info[i_chunk] = [i_chunk * self._n_items_in_chunk, (i_chunk + 1) * self._n_items_in_chunk]
+                return dataset[i_chunk * self._n_items_in_chunk: (i_chunk + 1) * self._n_items_in_chunk]
 
             return dataset[()]
 
-    def get_items_in_chunk(self):
-        return self._items_in_chunk
+    def get_chunks_info(self, i_chunk):
```

Comment:
This seems like the chunks are created before iterating, which is not necessary and may lead to some inconveniences. We should use an iterator to process chunks.